azuredevops/internal/service/policy/resource_branchpolicy_build_validation.go

```diff
@@ -61,6 +63,14 @@ func ResourceBranchPolicyBuildValidation() *schema.Resource {
 		Optional:     true,
 		ValidateFunc: validation.IntAtLeast(0),
 	}
+	settingsSchema[filenamePatterns] = &schema.Schema{
+		Type:     schema.TypeSet,
+		Optional: true,
+		Elem: &schema.Schema{
+			Type:         schema.TypeString,
+			ValidateFunc: validation.NoZeroValues,
+		},
```

Comment:
Please use `validate.NoEmptyStrings` because `validation.NoZeroValues` will successfully validate string that contain only whitespaces,